File: cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    float dollars;
    //define variable then get user input
   do
   {
        dollars = get_float("How much change is due?\n");
   }
   while (dollars <= 0);
    
    //round up
   int cents = round(dollars * 100);
   int coins = 0;
    
    //find number of coins needed
   while (cents >= 25)
   {
       cents = cents - 25;
       coins++;
   }
   while (cents >= 10)
   {
       cents = cents - 10;
       coins++;
   }
   while (cents >= 5)
   {
       cents = cents - 5;
       coins++;
   }
   while (cents >= 1)
   {
       cents = cents - 1;
       coins++;
   }

    //print result
   printf("Coins needed: %i\n", coins);

}

File: cash.py
from cs50 import get_float

#get positive change amount
while True:
    dollars = get_float("How much change is due?\n")
    if dollars > 0:
        break

#round dollars to cents
cents = int(dollars*100)

#use largest coins first
quarters = cents // 25
a = cents % 25

dimes = a // 10
b = a % 10

nickles = b // 5
c = b % 5

pennies = c // 1
d = c % 1

#keep track of coins used
coins = quarters + dimes + nickles + pennies

#print number of coins
print(coins)
